lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb

```diff
@@ -84,7 +84,7 @@ def create_xcconfig_file
             native_target.build_configurations.each do |configuration|
               path = target.xcconfig_path(configuration.name)
               gen = Generator::XCConfig::AggregateXCConfig.new(target, configuration.name)
-              gen.save_as(path)
+              update_changed_file(gen, path)
```

Comment:
my biggest annoyance in this PR is that we have to remember to call this method for every generated file we intend to produce.

This might be fine for now but I was hoping if someone can provide feedback on how to prevent this from happening.

Could all generators extend from a base class that instead takes care of this automatically?